x-pack/plugins/security_solution/public/management/cypress/tasks/login.ts

```diff
@@ -0,0 +1,336 @@
+
+/**
+ * The Kibana server endpoint used for authentication
+ */
+const LOGIN_API_ENDPOINT = '/internal/security/login';
+
+/**
+ * cy.visit will default to the baseUrl which uses the default kibana test user
+ * This function will override that functionality in cy.visit by building the baseUrl
+ * directly from the environment variables set up in x-pack/test/security_solution_cypress/runner.ts
+ *
+ * @param role string role/user to log in with
+ * @param route string route to visit
+ */
+export const getUrlWithRoute = (role: ROLE, route: string) => {
+  const url = Cypress.config().baseUrl;
+  const kibana = new URL(String(url));
+  const theUrl = `${Url.format({
+    auth: `${role}:changeme`,
+    username: role,
+    password: 'changeme',
+    protocol: kibana.protocol.replace(':', ''),
+    hostname: kibana.hostname,
+    port: kibana.port,
+  } as UrlObject)}${route.startsWith('/') ? '' : '/'}${route}`;
+  cy.log(`origin: ${theUrl}`);
+
+  return theUrl;
+};
+
+interface User {
+  username: string;
+  password: string;
+}
+
+/**
+ * Builds a URL with basic auth using the passed in user.
+ *
+ * @param user the user information to build the basic auth with
+ * @param route string route to visit
+ */
+export const constructUrlWithUser = (user: User, route: string) => {
+  const url = Cypress.config().baseUrl;
+  const kibana = new URL(String(url));
+  const hostname = kibana.hostname;
+  const username = user.username;
+  const password = user.password;
+  const protocol = kibana.protocol.replace(':', '');
+  const port = kibana.port;
+
+  const path = `${route.startsWith('/') ? '' : '/'}${route}`;
+  const strUrl = `${protocol}://${username}:${password}@${hostname}:${port}${path}`;
+  const builtUrl = new URL(strUrl);
+
+  cy.log(`origin: ${builtUrl.href}`);
+
+  return builtUrl.href;
+};
+
+export const getCurlScriptEnvVars = () => ({
+  ELASTICSEARCH_URL: Cypress.env('ELASTICSEARCH_URL'),
+  ELASTICSEARCH_USERNAME: Cypress.env('ELASTICSEARCH_USERNAME'),
+  ELASTICSEARCH_PASSWORD: Cypress.env('ELASTICSEARCH_PASSWORD'),
+  KIBANA_URL: Cypress.config().baseUrl,
+});
+
+export const postRoleAndUser = (role: ROLE) => {
+  const env = getCurlScriptEnvVars();
+
+  // post the role
+  cy.request({
+    method: 'PUT',
+    url: `${env.KIBANA_URL}/api/security/role/${role}`,
+    body: rolesMapping[role],
+    headers: {
+      'kbn-xsrf': 'cypress',
+    },
+    auth: {
+      user: 'elastic',
+      pass: 'changeme',
+    },
+  });
+
+  // post the user associated with the role to elasticsearch
+  cy.request({
+    method: 'POST',
+    url: `${env.KIBANA_URL}/internal/security/users/${role}`,
+    headers: {
+      'kbn-xsrf': 'cypress',
+    },
+    body: {
+      username: role,
+      password: 'changeme',
+      roles: [role],
+    },
+    auth: {
+      user: 'elastic',
+      pass: 'changeme',
+    },
+  });
+};
+
+export const deleteRoleAndUser = (role: ROLE) => {
+  const env = getCurlScriptEnvVars();
+
+  cy.request({
```

Comment:
Didn't know you can delete the role first even though a user has that role. If not, consider deleting the user first and then the role.